packages/default-theme/components/forms/SwPassword.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+<div class="sw-password">
+  <slot>
+
+    <slot name="message" :user="user">
+      <p class="message">
+        If you want to change the password to access your account, enter the
+        following information:
+      </p>
+    </slot>
+
+    <SfAlert
+      v-if="error"
+      class="sw-personal-info__alert"
+      type="danger"
+      message="Errors in the form"
+    />
+
+    <div class="sw-password__form form">
+      <slot name="form">
+        <div class="form">
+          <SfInput
+            v-model="password"
+            :valid="!$v.password.$error"
+            error-message="Current password is required"
+            type="password"
+            name="currentPassword"
+            label="Current Password"
+            class="form__element"
+            @blur="$v.password.$touch()"
+          />
+          <SfInput
+            v-model="newPassword"
+            :valid="!$v.newPassword.$error"
+            error-message="This field is required"
+            type="password"
+            name="newPassword"
+            label="New Password"
+            class="form__element form__element--half"
+            @blur="$v.newPassword.$touch()"
+          />
+          <SfInput
+            v-model="newPasswordConfirm"
+            :valid="!$v.newPasswordConfirm.$error"
+            error-message="This filed must be same as new password"
+            type="password"
+            name="repeatPassword"
+            label="Repeat Password"
+            class="form__element form__element--half form__element--half-even"
+            @blur="$v.newPasswordConfirm.$touch()"
+          />
+          <SfButton 
+            class="form__button" 
+            :disabled="$v.$invalid"
+            @click="invokeUpdate">
+            Update password
+          </SfButton>
+        </div>
+      </slot>
+    </div>  
+
+  </slot>
+</div>  
+  
+</template>
+
+<script>
+import { validationMixin } from 'vuelidate'
+import { required, minLength, sameAs } from 'vuelidate/lib/validators'
+import { SfInput, SfButton, SfAlert } from '@storefront-ui/vue'
+import { useUser } from '@shopware-pwa/composables'
+
+export default {
+  name: 'SwPassword',
+  components: { SfInput, SfButton, SfAlert },
+  mixins: [validationMixin],
+  props: {},
+  setup() {
+    const { user, error, updatePassword, refreshUser } = useUser()
+    return {
+      refreshUser,
+      updatePassword,
+      user,
+      error
+    }
+  },
+  data() {
+    return {
+      password: '',
+      newPassword: '',
+      newPasswordConfirm: '',
+      email: this.user && this.user.email,
+    }
+  },
+  methods: {
+    async invokeUpdate() {
+      const passwordChanged = await this.updatePassword({
+        password: this.password,
+        newPassword: this.newPassword,
+        newPasswordConfirm: this.newPasswordConfirm
+      })
+      await refreshUser()
+    }
+  },
+  validations: {
+    password: {
+      required,
+      minLenght: minLength(8)
```

Comment:
we should consider moving all the validation rules into some global config file or something. We can make another issue to cover that later on.